core/unit_test/TestReduce.hpp

```diff
@@ -720,6 +720,9 @@ TEST(TEST_CATEGORY, reduction_identity_min_max_floating_point_types) {
   // nvhpc on device doesn't use the correct neutral value for the min and max
   // reducers
 #if !(defined(KOKKOS_COMPILER_NVHPC) && defined(KOKKOS_ENABLE_OPENACC))
+#if defined(__FINITE_MATH_ONLY__) && __FINITE_MATH_ONLY__ > 0
+  GTEST_SKIP() << "skipping when assuming finite math";
+#endif
```

Comment:
Readability: don't nest it into the other guard, move it L720

Reply:
BTW the other guard L722 is misplaced and should be encompassing the test definition, i.e. start L722